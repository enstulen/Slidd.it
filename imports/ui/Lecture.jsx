import { FlowRouter } from 'meteor/kadira:flow-router';

import { Meteor } from 'meteor/meteor';
import React, { Component, PropTypes } from 'react';
import { CurrentUser } from '../startup/xUser.js';

export default class Lecture extends Component {
  constructor(props) {
    super(props);
    // Bind events
    this.handleButtonPress = this.handleButtonPress.bind(this);
  }

  handleButtonPress(event) {
    // Go to /lecture/<lectureName>
    event.preventDefault();
    const lectureName = this.props.lecture.lectureName;
    const params = {
      lectureName,
    };
    const routeName = 'lecture';
    FlowRouter.go(routeName, params, {});

    // set the user's current site to be the correct lecture
    Meteor.call('sliderValues.updateLecture', CurrentUser.state.userID, this.props.lecture.lectureName);
  }


  render() {
    if (this.props.frontPage) {
      return (
        <tr>
          <td className="left"><h4>{this.props.lecture.lectureName}</h4> </td>
          <td className="right"><button onClick={this.handleButtonPress} className="btn btn-success btn-lg">Select</button></td>
        </tr>
      );
    }
    return (
      <li><button className="liButton" onClick={this.handleButtonPress}>{this.props.lecture.lectureName}</button></li>
    );
  }
}

// // // // // PropTypes // // // // //

Lecture.defaultProps = {
  frontPage: false,
};

Lecture.propTypes = {
  frontPage: PropTypes.bool,
  lecture: PropTypes.object.isRequired,
};
