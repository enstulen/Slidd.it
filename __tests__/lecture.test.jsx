import renderer from 'react-test-renderer';
import React from 'react';
import { mount, shallow } from 'enzyme';
import Lecture from '../imports/ui/Lecture.jsx';

describe('Lecture component', () => {
  it('Renders correctly', () => {
    const rendered = renderer.create(
      <Lecture lecture={[]} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
  it('Correct length frontPage', () => {
    const wrapper = shallow(<Lecture lecture={[]} frontPage={true} />);
    expect(wrapper.find('.left').length).toEqual(1);
    expect(wrapper.find('.right').length).toEqual(1);
  });
  it('Correct length frontPage', () => {
    const wrapper = shallow(<Lecture lecture={[]} frontPage={false} />);
    expect(wrapper.find('.liButton').length).toEqual(1);
  });
});
