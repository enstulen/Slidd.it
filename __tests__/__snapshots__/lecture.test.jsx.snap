// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lecture component Renders correctly 1`] = `
<li>
  <button
    className="liButton"
    onClick={[Function]}
  />
</li>
`;
